Queue.c

```diff
@@ -0,0 +1,150 @@
+#include <mqueue.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+
+/* MACROs definition */
+#define SERVER_NAME "/server"
+#define BUF_SIZE    (1<<13)
+
+/* global variable declaration */
+static int token = 0;
+
+int main( int argc, char* argv[])
+{
+	int rc;
+    mqd_t sv, cl;
+    char buf_out[BUF_SIZE];
+    char buf_in[BUF_SIZE] ;
+    char cl_name[32];
+
+    /* initialise buffers to 0 */
```

Comment:
If you let the buffer reset here at the beginning of the program they will be reseted only once not before each reuse of them.
Better move the memset in the code block where they are used.